src/Core/Managed/Shared/Extensibility/Implementation/TaskTimer.cs

```diff
@@ -62,34 +72,50 @@ public bool IsStarted
         /// <param name="elapsed">The task to run.</param>
         public void Start(Func<Task> elapsed)
         {
-            var newTokenSource = new CancellationTokenSource();
+            DelayedWork delayedWork = new DelayedWork(elapsed);
+            DelayedWork previousWork = Interlocked.Exchange(ref this.latestWork, delayedWork);
+
+            if (previousWork != null)
+            {
+                previousWork.IsCancellationRequested = true;
```

Comment:
Thanks for the info.
I do not see any significant concern with this. Even at say, 5K req/sec we would thus get 300 tasks. Let's say this is the <1000 order of magnitude. These are 1000 objects that are already in memory anyway. These tasks will be scheduled for processing, a simple flag will be examined and then no operation will occur (the queue will be cleaned up). Notable is that all this is happening already. The difference is only in when all this occurs (at cancellation time vs when the tie interval expires). We do this now a little later, but we save all the allocations for the cancellation signals.

Normal perf runs will point out if we are missing something, but I do not see any need for any specialized testing. Agreed?